src/lib.rs

```diff
@@ -182,50 +182,69 @@ pub mod mock;
 //
 // can't use both sync and async secret service
 //
-#[cfg(all(feature = "sync-secret-service", feature = "async-secret-service"))]
+#[cfg(any(
+    all(feature = "sync-secret-service", feature = "async-secret-service"),
+    all(
+        feature = "sync-persistent-keyutils",
+        feature = "async-persistent-keyutils",
+    )
+))]
 compile_error!("This crate cannot use the secret-service both synchronously and asynchronously");
 
 //
 // pick the *nix keystore
 //
-
-#[cfg(all(target_os = "linux", feature = "linux-native"))]
+#[cfg(all(target_os = "linux", feature = "keyutils"))]
 pub mod keyutils;
-// use keyutils as default if secret-service is not available
 #[cfg(all(
     target_os = "linux",
     feature = "linux-native",
-    not(any(feature = "sync-secret-service", feature = "async-secret-service"))
+    not(feature = "sync-secret-service"),
+    not(feature = "async-secret-service"),
 ))]
 pub use keyutils as default;
 
 #[cfg(all(
     any(target_os = "linux", target_os = "freebsd", target_os = "openbsd"),
-    any(feature = "sync-secret-service", feature = "async-secret-service")
+    any(feature = "sync-secret-service", feature = "async-secret-service"),
 ))]
 pub mod secret_service;
-// use secret-service as default if it's available
 #[cfg(all(
     any(target_os = "linux", target_os = "freebsd", target_os = "openbsd"),
     any(feature = "sync-secret-service", feature = "async-secret-service"),
+    not(any(
+        feature = "sync-persistent-keyutils",
+        feature = "async-persistent-keyutils",
+    )),
 ))]
 pub use secret_service as default;
 
-// fallback to mock if neither keyutils nor secret service is available
-#[cfg(any(
-    all(
-        target_os = "linux",
-        not(any(
-            feature = "linux-native",
-            feature = "sync-secret-service",
-            feature = "async-secret-service"
-        ))
-    ),
-    all(
-        any(target_os = "freebsd", target_os = "openbsd"),
-        not(any(feature = "sync-secret-service", feature = "async-secret-service"))
+#[cfg(all(
+    target_os = "linux",
+    any(
+        feature = "sync-persistent-keyutils",
+        feature = "async-persistent-keyutils",
     )
 ))]
+pub mod keyutils_persistent;
+#[cfg(all(
+    target_os = "linux",
+    any(
+        feature = "sync-persistent-keyutils",
+        feature = "async-persistent-keyutils",
+    ),
+))]
+pub use keyutils_persistent as default;
+
+// fallback to mock if neither keyutils nor secret service is available
+#[cfg(all(
+    any(target_os = "linux", target_os = "freebsd", target_os = "openbsd"),
+    not(any(
+        feature = "keyutils",
+        feature = "sync-secret-service",
+        feature = "async-secret-service",
+    )),
+))]
```

Comment:
Revert this back to linux-native

Reply:
Wait, it will not work as expected if someone add the `linux-native` and one of the secret service?

Reply:
I confirm it does not work, see <https://github.com/soywod/keyring-rs/actions/runs/11353166029/job/31577538792>. I revert the changes.